ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java

```diff
@@ -64,6 +67,30 @@
     private final EventRunner eventRunner;
     private final FileSystemHelper fileSystemHelper;
 
+    private final Function<ListBucketResult, ImmutableList<FileSystemKey>> getFileSystemKeysFunction = new Function<ListBucketResult, ImmutableList<FileSystemKey>>() {
+        @Nullable
+        @Override
+        public ImmutableList<FileSystemKey> apply(@Nullable final ListBucketResult result) {
+            final ImmutableList.Builder<FileSystemKey> fileSystemKeys = new ImmutableList.Builder<>();
```

Comment:
You have another option for implementing this that doesn't require a data copy.  You could create two fluent iterables, on for each of the lists, apply a map function to each fluent iterable, and then concatenate the two fluent iterables together.